### ISSUE TITLE
Bun Shell crash when run tar and scp: Invalid state

### ISSUE THREAD

**Issue by ATQQ** [oven-sh/bun#8572]
### What version of Bun is running?

1.0.25+a8ff7be64

### What platform is your computer?

Darwin 22.4.0 arm64 arm

### What steps can reproduce the bug?

my shell main part

```sh
await $`pnpm buildTheme`

await $`echo ==🔧  ==`
await $`cd packages/theme/docs/.vitepress && tar -zvcf ${compressPkgName} dist && rm -rf dist && mv ${compressPkgName} ./../../../../`

await $`echo ==🚀  ==`
await $`scp ${compressPkgName} ${user}@${origin}:./`
await $`rm -rf ${compressPkgName}`

await $`echo ==✅  ==`
await $`ssh -p22 ${user}@${origin} "tar -xf ${compressPkgName} -C ${baseServerDir}/${fullOrigin}/${destDir}"`
```

![image](https://github.com/oven-sh/bun/assets/42485491/4b684a6c-1252-43ce-b0ee-3984858a4dba)

### What is the expected behavior?

run all shell success

### What do you see instead?

```sh
uh-oh: Invalid state
bun will crash now 😭😭😭

----- bun meta -----
Bun v1.0.25 (a8ff7be6) macOS Silicon 22.4.0
AutoCommand: 
Elapsed: 21339ms | User: 19ms | Sys: 21ms
RSS: 21.66MB | Peak: 23.66MB | Commit: 35.68MB | Faults: 0
----- bun meta -----

0   0x10163abfc
1   ???
2   ???

Search GitHub issues https://bun.sh/issues or ask for #help in https://bun.sh/discord

thread 101018 panic: Invalid state
zsh: abort      bun deploy-theme.ts
```

### Additional information

I split this script into two mini-scripts can run success

setp1 shell
```sh
await $`pnpm buildTheme`

await $`echo ==🔧  ==`
await $`cd packages/theme/docs/.vitepress && tar -zvcf ${compressPkgName} dist && rm -rf dist && mv ${compressPkgName} ./../../../../`
```
step2 shell
```sh
await $`echo ==🚀  ==`
await $`scp ${compressPkgName} ${user}@${origin}:./`
await $`rm -rf ${compressPkgName}`

await $`echo ==✅  ==`
await $`ssh -p22 ${user}@${origin} "tar -xf ${compressPkgName} -C ${baseServerDir}/${fullOrigin}/${destDir}"`
```

**Comment by zackradisic** [comment]
Can you provide more complete code to repro this?

**Comment by ATQQ** [comment]
ok

---
repo: https://github.com/ATQQ/sugar-blog
branch：[feature/bun-shell-test](https://github.com/ATQQ/sugar-blog/tree/feature/bun-shell-test)

```sh
# step 1
git clone https://github.com/ATQQ/sugar-blog.git
git checkout feature/bun-shell-test

# step2
pnpm install

# step3
pnpm buildlib

# step4
bun run deploy.ts
```
run in vs code shell

You can reproduce the question

![image](https://github.com/oven-sh/bun/assets/42485491/8c722776-547f-4bf6-89d6-4d26dcea3b24)

**Comment by dylan-conway** [comment]
This is fixed in bun v1.1.20
